### ISSUE TITLE
One file with multiple defmodule has no IntelliSense starting with the 2nd module in the file

### ISSUE THREAD

**Issue by aiwaiwa** [elixir-lsp/vscode-elixir-ls#334]
### Is this the right repo?

Not yet sure. 

### Environment

* Elixir & Erlang versions (elixir --version): Erlang/OTP 25 [erts-13.0.4] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit:ns]

Elixir 1.14.4 (compiled with Erlang/OTP 25)

* VSCode ElixirLS version: 0.14.7
* Operating System Version: win32 10.0.22621

### Troubleshooting

- [X] Restart your editor (which will restart ElixirLS) sometimes fixes issues
- [X] Stop your editor, remove the entire `.elixir_ls` directory, then restart your editor

###  Description

Any more than one `defmodule` in one source file at the root level get advantage of LSP discovery of any symbols other than basics like built-in macros.

In other words
```
# File A
defmodule A.A

defmodule A.B
  # Will not IntelliSense
```
Internals of `A.B` seem to lack any features of IntelliSense.

Backwards they seem to be discovered and navigated without issues though, again provided those are not coming from the 2nd, 3rd, etc. defmodules:

```
# File B

defmodule B.A
  # Will IntelliSense A.A and A.B.
defmodule B.B
  # Will not IntelliSense
```

**Comment by lukaszsamson** [comment]
@aiwaiwa Can you provide a piece of correct code that reproduces this? The one above is not parsable.

I'm seeing this
<img width="673" alt="Screenshot 2023-05-29 at 15 04 06" src="https://github.com/elixir-lsp/vscode-elixir-ls/assets/1078186/2d596cef-4286-42d2-99bf-42b46d7fdff2">

**Comment by aiwaiwa** [comment]
I must admit, it only shows under certain circumstances which I can't yet narrow down.

https://github.com/aiwaiwa/sample_phoenix_project

Navigate to `lib/sample_phoenix_project_web/live/sample/index.ex`, to the second defmodule.

**Comment by aiwaiwa** [comment]
Reporting this as still relevant after updating to `0.15.0` (thank you @lukaszsamson, you are amazing!)

The sample project reports this when attempting to navigate to any symbols in `lib/sample_phoenix_project_web/live/sample/index.ex` (although only second module does not provide any IntelliSense, the `Warn` is always issued):

```
[Warn  - 7:32:24 AM] ** (ErlangError) Erlang error: "CompileError during metadata build pre:\nnofile: module SamplePhoenixProjectWeb.Sample.Struct.SamplePhoenixProjectWeb.Sample.Struct.SamplePhoenixProjectWeb is not loaded and could not be found\nast node: {:use, [line: 12, column: 3], [{:__aliases__, [line: 12, column: 7], [:SamplePhoenixProjectWeb]}, :live_view]}"
    (elixir 1.14.4) src/elixir_expand.erl:90: :elixir_expand.expand/3
    (elixir 1.14.4) src/elixir_expand.erl:536: :elixir_expand.expand_block/5
    (elixir 1.14.4) src/elixir_expand.erl:40: :elixir_expand.expand/3
    (elixir 1.14.4) src/elixir.erl:376: :elixir.quoted_to_erl/4
    (elixir 1.14.4) src/elixir.erl:277: :elixir.eval_forms/4
    (elixir 1.14.4) lib/module/parallel_checker.ex:110: Module.ParallelChecker.verify/1
    (elixir 1.14.4) lib/code.ex:425: Code.validated_eval_string/3
```

Removing `.elixir_ls`, restarting LS, leads to the same outcome.
